### PR TITLE
boards/samr34-xpro: add minimal support for WLR089 Xplained Pro

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The [ WLR089 Xplained Pro](https://www.microchip.com/en-us/development-tool/EV23M25A)  is pretty much identical to the [SAM R34 Xplained Pro](https://www.microchip.com/en-us/development-tool/DM320111) - this adds a `doc.txt` to clarify that.

The only notable difference so far is, that the device ID returned by the sx1276 radio is `0x13` instead of `0x12`.
It works when adjusting the check accordingly.

### Testing procedure

@firas-hamdi ran `examples/gnrc_lorawan` and `tests/driver_sx127x`

### CODE REVIEW

**Comment by aabadie** [comment]
I don't see the point in modifying the samr34-xpro board documentation only. This board already provides an integrated sx1276 chip that already works.
The WLR089 extension could work on any xpro board I think.

**Comment by benpicco** [comment]
The WLR089 is not an extension, it is just a SAM R34 packaged as a FCC certified module.

File: boards/samr34-xpro/doc.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/**
+@defgroup    boards_samr34-xpro Microchip SAM R34 Xplained Pro
+@ingroup     boards
+@brief       Support for the Microchip SAM R34 Xplained Pro and WLR089 Xplained Pro
+
+## Overview
+
+The `SAMR34 Xplained Pro` is a compact evaluation board by Atmel featuring a
```
Comment (dylad) [inline boards/samr34-xpro/doc.txt:8]:
```suggestion
The `SAMR34 Xplained Pro` is a compact evaluation board by Microchip featuring a
```

**Review by dylad — COMMENTED** [review]
LGTM.

**Comment by aabadie** [comment]
> The WLR089 is not an extension, it is just a SAM R34 packaged as a FCC certified module.

Ok. But then the new define shouldn't be suffixed with `R34`, it's confusing. Better use `WLR089` or something like this. But it should mention the radio module that provides this special case.

**Review by aabadie — APPROVED** [review]
ACK

**Pull request merged**